### ISSUE TITLE
Xcode 16 and iOS 18 project not compiling

### ISSUE THREAD

**Issue by RensithUdara** [flutter/flutter#157461]
I updated Xcode and MacOS recently and haven't been able to compile my Flutter app on iOS devices/simulators since then. The error keeps changing every time I run it but here's one of the output in the terminal after running flutter run:

Launching lib/main.dart on iPhone 15 Pro Max in debug mode... Running pod install... 17.4s Running Xcode build...
Xcode build done. 40.9s Failed to build iOS app Error (Xcode): no such file or directory: '/Users/rensithudaragonalagoda/Library/Developer/Xcode/DerivedData/ModuleCache.noindex/Session.modulevalidation'

Error (Xcode): stat cache file '/Users/rensithudaragonalagoda/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator18.0-22A3362-db63dc9361471f152f572502bdbfe70a.sdkstatcache' not found

Could not build the application for the simulator. Error launching application on iPhone 15 Pro Max.

What I have tried so far:
- Deleting iOS SDK and simulators
- Cleaning Xcode build cache using cmd+shift+k
- Creating a new Flutter project and trying to compile it (it failed so that reduces the chances of it being a Flutter issue)
- Clearing the DerivedData folder
- Clearing settings and cache of simulators
- Restarting laptop

Versions:

- Xcode: 16.0
- iOS: 18.0
- MacOS: 15.1 (didn't work with 15.0 either)
- Flutter: 3.24

**Comment by huycozy** [comment]
Can you attempt to check on an iOS native project (default one) and run it on iOS simulator to see if the same error occurs there?

If the issue only appears on Flutter project, please provide the entire output of `flutter run -v` [^1]. Thanks!

[^1]: The output may be too large to be uploaded to GitHub, you may upload it as a txt file or use online tools like https://pastebin.com/ to share it.

**Comment by RensithUdara** [comment]
rensithudaragonalagoda@Rensiths-MacBook-Air selfcare % flutter doctor -v
[✓] Flutter (Channel stable, 3.24.3, on macOS 15.0 24A5320a darwin-arm64, locale en-LK)
    • Flutter version 3.24.3 on channel stable at /Users/rensithudaragonalagoda/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2663184aa7 (6 weeks ago), 2024-09-11 16:27:48 -0500
    • Engine revision 36335019a8
    • Dart version 3.5.3
    • DevTools version 2.37.3

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/rensithudaragonalagoda/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/to/macos-android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.94.2)
    • VS Code at /Users/rensithudaragonalagoda/Desktop/Visual Studio Code.app/Contents
    • Flutter extension version 3.98.0

[✓] VS Code (version 1.94.2)
    • VS Code at /Users/rensithudaragonalagoda/Downloads/Visual Studio Code.app/Contents
    • Flutter extension version 3.98.0

[✓] Connected device (4 available)
    • New IPhone (mobile)             • 2E660FC9-D872-4EB1-88C0-B09E1F63DA8C • ios            • com.apple.CoreSimulator.SimRuntime.iOS-18-1 (simulator)
    • macOS (desktop)                 • macos                                • darwin-arm64   • macOS 15.0 24A5320a darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                • darwin         • macOS 15.0 24A5320a darwin-arm64
    • Chrome (web)                    • chrome                               • web-javascript • Google Chrome 130.0.6723.69
    ! Error: Browsing on the local area network for Rensith Udara’s iPhone. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this
      Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
rensithudaragonalagoda@Rensiths-MacBook-Air selfcare %

**Comment by chiragbhansali** [comment]
> Can you attempt to check on an iOS native project (default one) and run it on iOS simulator to see if the same error occurs there?
> 
> If the issue only appears on Flutter project, please provide the entire output of `flutter run -v` [1](#user-content-fn-1-f3cd43b14c4a77d099ee27c8b48d8a24). Thanks!
> 
> ## Footnotes
> 1. The output may be too large to be uploaded to GitHub, you may upload it as a txt file or use online tools like https://pastebin.com/ to share it. [↩](#user-content-fnref-1-f3cd43b14c4a77d099ee27c8b48d8a24)

I faced the same problem and it exists even for native iOS projects. I did find a workaround though: change the location of DerivedData folder in Xcode Workspace settings (File -> Workspace Settings) to something else.

**Comment by tuanhwing** [comment]
I faced the same problem.
A few days ago I was still able to build. But today when I updated Xcode 16.1 RC and downgraded to Xcode 16.0, this problem appeared. 

My Derived Data configured like bellow:
- Xcode -> File -> Workspace Settings : Default Location 
- Xcode -> Settings -> Locations : Default

How to solve it?

**Comment by huycozy** [comment]
As @chiragbhansali pointed out, this looks like an issue with Xcode in local environment. @RensithUdara, @tuanhwing: Let's try changing to another location to see if it helps.

**Comment by Gufdoor** [comment]
Same here. I was using Xcode 15 and iOS 17.5 simulators, everything worked fine. After updating to Xcode 16 and iOS 18 simulators, I can only build apps on Android.  Now every build generates this (summarised) exception: 

```Error output from Xcode build:
↳
    --- xcodebuild: WARNING: Using the first of multiple matching destinations:
    { platform:iOS Simulator, id:69C6AA86-1929-4132-84A0-A1809E079FBA, OS:18.0, name:iPhone SE (3rd generation) }
    { platform:iOS Simulator, id:69C6AA86-1929-4132-84A0-A1809E079FBA, OS:18.0, name:iPhone SE (3rd generation) }
    ** BUILD FAILED **
```

Apparently, in this [thread](https://forums.developer.apple.com/forums/thread/764196), some people found a solution by updating Xcode to 16 Beta 2, but still the actual error wasn't found.

**Comment by tuanhwing** [comment]
> thread

Thanks @huycozy . It worked for me

**Comment by RensithUdara** [comment]
@huycozy It worked. Thank you

**Comment by huycozy** [comment]
Thanks for confirming this. Since this is an issue with Xcode in local environment, I will close this issue. Ref: https://github.com/flutter/flutter/issues/157461#issuecomment-2435204108

**Comment by sanderpensa** [comment]
I think it's something related to bad/damaged Xcode install. I managed to get Xcode 16.0 iOS 18.0 simulator working again when I uninstalled all related Xcode caches, folders, images, etc... mostly following this guide https://dzone.com/articles/how-to-completely-uninstall-xcode-from-mac.

What I did was removing ALL Xcode related files `mdfind -name "xcode"` that were not project related and not under system permissions meaning all files that `sudo rm -rf` could remove that were somehow related to Xcode.

After that the new install of Xcode did work again. I was using https://github.com/XcodesOrg/xcodes version manager to install Xcode 16.0.

NOTE: The custom DerivedData folder assignment worked also, however it's not a great solution if there are multiple developers. It's an easy workaround vs clean install.
